src/main.rs

```diff
@@ -3,21 +3,40 @@
 
 //! parsec-tool: a tool for interfacing with the Parsec service from the command-line.
 
+use parsec_client::core::ipc_handler::unix_socket::Handler;
 use parsec_client::BasicClient;
-use parsec_tool::cli;
 use parsec_tool::common::PROJECT_NAME;
-use parsec_tool::err;
+use parsec_tool::{cli, err, error};
 use std::convert::TryInto;
+use std::env;
+use std::io::{Error, ErrorKind};
+use std::path::PathBuf;
+use std::time::Duration;
 use structopt::StructOpt;
 
-fn main() -> std::io::Result<()> {
+const TIMEOUT: Option<Duration> = Some(Duration::from_secs(30));
+
+fn main() -> Result<(), error::Error> {
     env_logger::init();
 
     let matches = cli::ParsecToolApp::from_args();
-
-    let mut client = BasicClient::new(Some(PROJECT_NAME.to_string())).map_err(|e| {
+    let mut client = BasicClient::new_naked();
```

Comment:
Could you keep the line above but just call `set_ipc_handler` after? To not have to set the default provider + auth

Reply:
Well, how would you then get the default values if you don't know where the socket is? 😛

Reply:
ah yes 🤡 
hmm I am wondering if we should not just go ahead and implement parallaxsecond/parsec-client-rust#37 in the client? I think we will need the feature anyway and that way the code here will stay the same? We would only need to document in the `README.md` or somewhere that this can be used to change the socket path.
Then we can see later if there is a need to also change it via the command line?

Reply:
Not sure, I'd wait for Paul on that one - I say we merge this however we agree on in this PR, and then we can decide about the client feature later. The environment variable mechanism would essentially be part of the client API so we can't backtrack that easily on it, though it seems quite straight forward how it would work